packages/datetime/src/datePicker.tsx

```diff
@@ -176,7 +182,7 @@ export class DatePicker extends AbstractPureComponent2<IDatePickerProps, IDatePi
                         onMonthChange={this.handleMonthChange}
                         selectedDays={this.state.value}
                         toMonth={maxDate}
-                        renderDay={this.renderDay}
+                        renderDay={dayPickerProps?.renderDay ? dayPickerProps?.renderDay : this.renderDay}
```

Comment:
you can simplify this, like I suggested in https://github.com/palantir/blueprint/issues/3996#issuecomment-598431511... that's what `??` is for

```suggestion
                        renderDay={dayPickerProps?.renderDay ?? this.renderDay}
```